src/sagas/task.saga.test.js

```diff
@@ -0,0 +1,390 @@
+
+      // `take` a log message
+      clone.next();
+
+      expect(
+        clone.next({ channel: 'stderr', text, isDevServerFail: true }).value
+      ).toEqual(put(receiveDataFromTaskExecution(task, text, true)));
+    });
+
+    it('should display logs from stderr', () => {
+      const clone = saga.clone();
+      const text = "Oops, something went wrong but it's probably not fatal";
+
+      // `take` a log message
+      clone.next();
+
+      expect(clone.next({ channel: 'stdout', text }).value).toEqual(
+        put(receiveDataFromTaskExecution(task, text))
+      );
+    });
+
+    it('should complete on exit', () => {
+      const clone = saga.clone();
+      const timestamp = new Date();
+
+      // `take` a log message
+      clone.next();
+
+      expect(
+        clone.next({ channel: 'exit', timestamp, wasSuccessful: true }).value
+      ).toEqual(call(displayTaskComplete, task, true));
+
+      expect(clone.next().value).toEqual(
+        put(completeTask(task, timestamp, true))
+      );
+    });
+  });
+
+  describe('taskRun saga', () => {
+    const projectReact = { type: 'create-react-app' };
+    const projectGatsby = { type: 'gatsby' };
+    const projectPath = '/path/to/project';
+    const processId = 12345;
+    const mockProcess = mockSpawn(processId);
+
+    it('should add --coverage arg to create-react-app tests', () => {
+      const name = 'test';
+      const saga = taskRun({ task: { name, projectId: '' } });
+      saga.next();
+      saga.next(projectReact);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name, '--coverage'],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    it('should run without additionalArgs for gatsby', () => {
+      const name = 'test';
+      const saga = taskRun({ task: { name, projectId: '' } });
+      saga.next();
+      saga.next(projectGatsby);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    it('should run without additional args for create-react-app non-test tasks', () => {
+      const name = 'not-test';
+      const saga = taskRun({ task: { name, projectId: '' } });
+      saga.next();
+      saga.next(projectReact);
+
+      expect(saga.next(projectPath).value).toEqual(
+        call(
+          [childProcess, childProcess.spawn],
+          PACKAGE_MANAGER_CMD,
+          ['run', name],
+          {
+            cwd: projectPath,
+            env: getBaseProjectEnvironment(projectPath),
+          }
+        )
+      );
+    });
+
+    describe('running', () => {
```

Comment:
I had to create this new block because of a `no-shadow` lint error; `saga` was being defined in the parent block, and then shadowed in the earlier tests like `should add --coverage arg to create-react-app tests`.

I think it's better anyway to have the tests that use a "parent" saga be defined in their own "describe". Although "running" is not a great name for that block. Open to suggestions!

Reply:
This looks great, and I agree, it does make more sense to group "child" saga tests.